internal/controller/appwrapper_controller.go

```diff
@@ -193,6 +215,17 @@ func (r *AppWrapperReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			// reset status to queued/idle
 			appWrapper.Status.Restarts += 1
 			appWrapper.Status.RequeueTimestamp = metav1.Now() // overwrite requeue decision time with completion time
+			msg := "Requeued by MCAD"
+			if decision, ok := r.Decisions[appWrapper.UID]; ok && decision.reason == mcadv1beta1.QueuedRequeue {
+				msg = fmt.Sprintf("Requeued because %s", decision.message)
+				delete(r.Decisions, appWrapper.UID)
```

Comment:
Should the deletion be moved after a successful status update?

Reply:
Fixed.